src/hotspot/share/runtime/arguments.cpp

```diff
@@ -585,26 +585,33 @@ static AliasedFlag const aliased_jvm_flags[] = {
-  { "TraceClassLoading",         LogLevel::Info,  true,  LOG_TAGS(class, load) },
-  { "TraceClassLoadingPreorder", LogLevel::Debug, true,  LOG_TAGS(class, preorder) },
-  { "TraceClassPaths",           LogLevel::Info,  true,  LOG_TAGS(class, path) },
-  { "TraceClassResolution",      LogLevel::Debug, true,  LOG_TAGS(class, resolve) },
-  { "TraceClassUnloading",       LogLevel::Info,  true,  LOG_TAGS(class, unload) },
-  { "TraceExceptions",           LogLevel::Info,  true,  LOG_TAGS(exceptions) },
-  { "TraceInvokeDynamic",        LogLevel::Debug, true,  LOG_TAGS(methodhandles, indy) },
-  { "TraceLoaderConstraints",    LogLevel::Info,  true,  LOG_TAGS(class, loader, constraints) },
-  { "TraceMethodHandles",        LogLevel::Info,  true,  LOG_TAGS(methodhandles) },
-  { "TraceMonitorInflation",     LogLevel::Trace, true,  LOG_TAGS(monitorinflation) },
-  { "TraceSafepointCleanupTime", LogLevel::Info,  true,  LOG_TAGS(safepoint, cleanup) },
-  { "TraceJVMTIObjectTagging",   LogLevel::Debug, true,  LOG_TAGS(jvmti, objecttagging) },
-  { "TraceRedefineClasses",      LogLevel::Info,  false, LOG_TAGS(redefine, class) },
-  { "PrintJNIResolving",         LogLevel::Debug, true,  LOG_TAGS(jni, resolve) },
   { NULL,                        LogLevel::Off,   false, LOG_TAGS(_NO_TAG) }
 };
 
+// Use this to obsolete old tracing flags and suggest logging alternative.
+static AliasedObsoleteLoggingFlag const removed_product_logging_flags[] = {
+  { "PrintSharedSpaces",         "-Xlog:cds=",                 "info",  "16.0" },
+  { "TraceBiasedLocking",        "-Xlog:biasedlocking=",       "info",  "16.0" },
+  { "TraceClassLoading",         "-Xlog:class+load=",          "info",  "16.0" },
+  { "TraceClassLoadingPreorder", "-Xlog:class+preorder=",      "debug", "16.0" },
+  { "TraceClassPaths",           "-Xlog:class+path=",          "info",  "16.0" },
+  { "TraceClassResolution",      "-Xlog:class+resolve=",       "debug", "16.0" },
+  { "TraceClassUnloading",       "-Xlog:class+unload=",        "info",  "16.0" },
+  { "TraceExceptions",           "-Xlog:exceptions=",          "info",  "16.0" },
+  { "TraceInvokeDynamic",        "-Xlog:methodhandles+indy=",  "debug", "16.0" },
+  { "TraceLoaderConstraints",    "-Xlog:loader+constraints=",  "info",  "16.0" },
+  { "TraceMethodHandles",        "-Xlog:methodhandles=",       "info",  "16.0" },
+  { "TraceMonitorInflation",     "-Xlog:monitorinflation=",    "trace", "16.0" },
+  { "TraceSafepointCleanupTime", "-Xlog:safepoint+cleanup=",   "info",  "16.0" },
+  { "TraceJVMTIObjectTagging",   "-Xlog:jvmti+objecttagging=", "debug", "16.0" },
+  { "TraceRedefineClasses",      "-Xlog:redefine+class=",      "info",  "16.0" },
+  { "PrintJNIResolving",         "-Xlog:jni+resolve=",         "debug", "16.0" },
+  { NULL, NULL, NULL, NULL }
+};
+
 #ifndef PRODUCT
 // These options are removed in jdk9. Remove this code for jdk10.
 static AliasedFlag const removed_develop_logging_flags[] = {
```

Comment:
I think this removed_develop_logging_flags infrastructure should be removed.

Reply:
done.